scss/mixins/_utilities.scss

```diff
@@ -41,22 +45,46 @@
       }
     }
 
+    $is-css-var: map-get($utility, css-var);
+    $is-local-vars: map-get($utility, local-vars);
     $is-rtl: map-get($utility, rtl);
 
     @if $value != null {
       @if $is-rtl == false {
         /* rtl:begin:remove */
       }
-      .#{$property-class + $infix + $property-class-modifier} {
-        @each $property in $properties {
-          #{$property}: $value if($enable-important-utilities, !important, null);
-          // Boosted mod: ensure contrasts in color utilities
-          @if "background-color" == $property and "transparent" != inspect($value) {
-            color: color-contrast($value);
-          } @else if "color" == $property and "inherit" != inspect($value) and $accessible-orange != $value {
-            background-color: color-contrast($value);
+
+      @if $is-css-var {
+        .#{$property-class + $infix + $property-class-modifier} {
+          --#{$variable-prefix}#{$property-class}: #{$value};
+        }
+
+        @each $pseudo in $state {
+          .#{$property-class + $infix + $property-class-modifier}-#{$pseudo}:#{$pseudo} {
+            --#{$variable-prefix}#{$property-class}: #{$value};
+          }
+        }
+      } @else {
+        .#{$property-class + $infix + $property-class-modifier} {
+          @each $property in $properties {
+            @if $is-local-vars {
+              @each $local-var, $value in $is-local-vars {
+                --#{$variable-prefix}#{$local-var}: #{$value};
+              }
+            }
+            #{$property}: $value if($enable-important-utilities, !important, null);
+
+            // Boosted mod: ensure contrasts in color utilities
+            // FIXME: color-contrast($value) throws an error
+            // @if "background-color" == $property and "transparent" != inspect($value) {
+            //  @debug "#{$property}: #{$value}";
+            //  color: color-contrast($value);
```

Comment:
Thanks @ffoodd.
I'm not sure it would work but I'll give it a try.
Indeed, if I'm not mistaken, there won't be corresponding Sass variable all the time:
* `var(--bs-primary-rgb)` -> `$primary`
* `var(--bs-body-rgb)` -> `$body-bg` (not the same name)

Reply:
When using the css variables, current solution can't be still valid. If the user change the css variable value (it has been done for that) then color contrast calculated during sass compilation can be wrong.
Maybe we need to give a try to pure css luminance calculation. I've found that interesting ressource : https://css-tricks.com/switch-font-color-for-different-backgrounds-with-css/ don't know what will be the effort to convert our variable and implement it

Reply:
It shouldn't matter, this check was done only for Orange brand colors. Boosted is not supposed to support any other color palette.

As a matter of fact, custom property names shouldn't matter either: simple check if it **contains** any of the theme colors, grab it, and get its contrasted color.

Reply:
Got a naive code that seems to work (and that could probably be improved) but I'm not sure to be in the right direction given your last comment @ffoodd.

```scss
// Boosted mod: ensure contrasts in color utilities
@if "background-color" == $property and "transparent" != inspect($value) {
  // @debug "#{$property}: #{$value}";
  @if type-of($value) == "string" {
    @if (str-index($value, "var(--bs-primary-rgb)")) {
      $value: $primary;
    }
    @else if (str-index($value, "var(--bs-secondary-rgb)")) {
      $value: $secondary;
    }
    @else if (str-index($value, "var(--bs-success-rgb)")) {
      $value: $success;
    }
    @else if (str-index($value, "var(--bs-info-rgb)")) {
      $value: $info;
    }
    @else if (str-index($value, "var(--bs-warning-rgb)")) {
      $value: $warning;
    }
    @else if (str-index($value, "var(--bs-danger-rgb)")) {
      $value: $danger;
    }
    @else if (str-index($value, "var(--bs-light-rgb)")) {
      $value: $light;
    }
    @else if (str-index($value, "var(--bs-white-rgb)")) {
      $value: $white;
    }
    @else if (str-index($value, "var(--bs-body-rgb)")) {
      $value: $body-bg;
    }
    @else if (str-index($value, "var(--bs-dark-rgb)")) {
      $value: $dark;
    }
    @else if (str-index($value, "var(--bs-black-rgb)")) {
      $value: $black;
    }
  }
  color: color-contrast($value);
} @else if "color" == $property and "inherit" != inspect($value) and $accessible-orange != $value and (type-of($value) == "string" and null == (str-index($value, "var(--bs-primary-rgb)"))) {
  // @debug "#{$property}: #{$value}";

  @if type-of($value) == "string" {
    @if (str-index($value, "var(--bs-white-rgb)")) {
      $value: $white;
    }
    @else if (str-index($value, "var(--bs-light-rgb)")) {
      $value: $light;
    }
    @else if (str-index($value, "var(--bs-body-rgb)")) {
      $value: $body-color;
    }
  }
  background-color: color-contrast($value);
}
// End mod
}
}
```

Reply:
Could be simplified indeed:
1. first extract the custom property name's middle key: `$name = str-slice($value, 10, -6)` to drop `var(--bs-` (p chars) and `-rgb)` (5chars).
2. then get the key from `$theme-colors` if it exists: `@if map-has-key($theme-colors, $name) { $value = $map-get($theme-colors, $name); }`.

Didn't test this out but you get the idea :)

Reply:
Simplified it this way: https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/772/commits/78df30909480922d4bb26d4d683beb5ce7424046